conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DefaultStaticFactoryMethodGenerator.java

```diff
@@ -159,41 +174,74 @@ private Optional<FieldSpec> serializer(EndpointName endpointName, Type type) {
                 .build());
     }
 
-    private Optional<FieldSpec> deserializer(EndpointName endpointName, Optional<Type> type) {
+    private Optional<FieldSpec> deserializer(
+            TypeName responseType, EndpointDefinition endpointDef, EndpointName endpointName, Optional<Type> type) {
```

Comment:
nit: no need for `endpointName` and `type` if we are passing the full `EndpointDefinition` here.

```suggestion
    private Optional<FieldSpec> deserializer(TypeName responseType, EndpointDefinition endpointDef) {
            EndpointName endpointName = endpointDef.getEndpointName();
            Optional<Type> type = endpointDef.getReturns();
```